pkg/utils/mainhelper/boot.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package mainhelper provides helper functions for common boilerplate code in
+// writing a main function such as setting up kube informers.
+package mainhelper
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/google/knative-gcp/pkg/observability"
+	"github.com/kelseyhightower/envconfig"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/client-go/kubernetes"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection/sharedmain"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/profiling"
+)
+
+// BootRes holds a collection of objects after booting for convenient access by
+// other custom logic in the main function.
+type BootRes struct {
+	Logger     *zap.SugaredLogger
+	KubeClient kubernetes.Interface
+	CMPWatcher configmap.Watcher
+	Cleanup    func()
+}
+
+// Similar to sharedmain.Main, Boot runs common logic in starting a main function,
+// it returns a result object that contains useful artifacts for later use.
+// Unlike sharedmain.Main, Boot is meant to be run as a helper function in any main
+// functions, while sharedmain.Main runs controllers with predefined method signatures.
+func Boot(component string, opts ...BootOption) (context.Context, *BootRes) {
```

Comment:
Just providing some naming idea: how about `Init()` like we have `google.Init()`

Reply:
Sure, changed to `Init`